x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -440,6 +483,11 @@ static Set<String> fieldNames(LogicalPlan parsed, Set<String> enrichPolicyMatchF
                 // The exact name of the field will be added later as part of enrichPolicyMatchFields Set
                 enrichRefs.removeIf(attr -> attr instanceof EmptyAttribute);
                 references.addAll(enrichRefs);
+            } else if (p instanceof LookupJoin join) {
+                keepJoinReferences.addAll(join.config().matchFields());  // TODO: why is this empty
```

Comment:
Had to have a look: the LookupJoin is [created with empty join fields](https://github.com/elastic/elasticsearch/blob/041022944e7df963d4a3269dfa6ec7791247251e/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/LookupJoin.java#L32), and the join fields are extracted from the `UsingJoinType` only later in the optimiser (when `surrogate()` is invoked) -- I suppose using the empty fields needs fixing (at some point before or after this PR?).

Reply:
After this PR, hence the TODO to remind us to fix this. I view this work as a first minimal fix to the hard-coded field mappings, and want it to be clean, simple and easy to review.